packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/cloudwatch-logs.ts

```diff
@@ -0,0 +1,100 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as awslogs from '@aws-cdk/aws-logs';
+import * as cdk from '@aws-cdk/core';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+/**
+ * Settings for the hook which mutates the application container
+ * to send logs to CloudWatch Logs
+ */
+export interface CloudWatchLogsProps {
+  /**
+   * The parent service that is being mutated
+   */
+  readonly parentService: Service;
+
+  /**
+   * The log group into which logs should be routed
+   */
+  readonly logGroup: awslogs.LogGroup;
+}
+
+/**
+ * This hook modifies the application container's settings so that
+ * it routes logs using FireLens
+ */
+export class CloudWatchLogsMutatingHook extends ContainerMutatingHook {
+  private parentService: Service;
+  private logGroup: awslogs.LogGroup;
+
+  constructor(props: CloudWatchLogsProps) {
+    super();
+    this.parentService = props.parentService;
+    this.logGroup = props.logGroup;
+  }
+
+  public mutateContainerDefinition(props: ecs.ContainerDefinitionOptions) {
+    return {
+      ...props,
+      logging: new ecs.AwsLogDriver({
+        streamPrefix: this.parentService.id,
+        logGroup: this.logGroup,
+      }),
+    } as ecs.ContainerDefinitionOptions;
+  }
+}
+
+/**
+ * Settings for the hook which mutates the application container
+ * to send logs to CloudWatch Logs
+ */
+export interface CloudWatchLogsExtensionProps {
+  /**
+   * The log group into which logs should be routed
+   */
+  readonly logGroup?: awslogs.LogGroup;
```

Comment:
Might want to make this mandatory since it's the only property in this interface